### PR TITLE
Fix KMAC docs

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/EVP_MAC.pod [hunk 1]
```diff
@@ -235,7 +235,7 @@ Note that not all algorithms may support all digests.
 HMAC does not support variable output length digests such as SHAKE128
 or SHAKE256.
 
-=item B<OSSL_MAC_PARAM_SIZE> ("size") <unsigned integer>
```
Comment (paulidale) [inline doc/man3/EVP_MAC.pod]: I think this one should stay _unsigned int_.  It is talking about the type of the PARAM not the C type.

File: doc/man7/EVP_MAC-KMAC.pod [hunk 2]
```diff
@@ -39,12 +39,12 @@ EVP_MAC_CTX_get_params(), or with EVP_MAC_size().
 
 =item B<OSSL_MAC_PARAM_SIZE> ("size") <size_t>
 
-=item B<OSSL_MAC_PARAM_XOF>
```
Comment (paulidale) [inline doc/man7/EVP_MAC-KMAC.pod]: This should be _int_, not _integer_.  These are C types.

File: doc/man1/openssl-mac.pod [hunk 3]
```diff
@@ -89,7 +89,7 @@ printable characters only).
 
 Used by GMAC to specify an IV in hexadecimal form (two hex digits per byte).
 
-=item B<outlen:int>
+=item B<size:size_t>
```
Comment (levitte) [inline doc/man1/openssl-mac.pod]: Why would someone reading the manual for a command care that the number is a `size_t`?

File: doc/man7/EVP_MAC-KMAC.pod (hunk 2 above)
Comment (levitte) [reply doc/man7/EVP_MAC-KMAC.pod]: No.  The *implementation* will most probably read it as an int, but from a param perspective, it's a `OSSL_PARAM_INTEGER`.

Or, seen differently, how is `octet string` a C type?

File: doc/man3/EVP_MAC.pod (hunk 1 above)
Comment (slontis) [reply doc/man3/EVP_MAC.pod]: There are others above this that use integer...

File: doc/man1/openssl-mac.pod (hunk 3 above)
Comment (slontis) [reply doc/man1/openssl-mac.pod]: changed back to int

File: doc/man7/EVP_MAC-KMAC.pod (hunk 2 above)
Comment (slontis) [reply doc/man7/EVP_MAC-KMAC.pod]: I am going to leave it as an integer, as that is what it uses in EVP_MAC.pod..

File: doc/man3/EVP_MAC.pod (hunk 1 above)
Comment (slontis) [reply doc/man3/EVP_MAC.pod]: It is a size_t in other files, so I am going to leave it as a size_t.

Comment (levitte) [reply doc/man3/EVP_MAC.pod]: It would be better to change the size_t in those other files.

Comment (slontis) [reply doc/man3/EVP_MAC.pod]: I really dont agree, and I think this is VERY confusing :)
If this is supposed to be what type the param is then its a size_t - not a unsigned integer. 
We are using octet string etc which corresponds to the type we use when we construct one of these params. 
We dont do construct_uint() for it.

Comment (levitte) [reply doc/man3/EVP_MAC.pod]: The parameter data type is OSSL_PARAM_UNSIGNED_INTEGER.  As long as that type as well as the data size is compatible, it doesn't really matter of the one end uses uint and the other uses size_t.

The `OSSL_PARAM_` functions are pickier, true, but they are not mandatory (or do you want them to be?)

Comment (slontis) [reply doc/man3/EVP_MAC.pod]: I am thinking of the end user - not how you have implemented it under the hood.

It should be clear to a user that you need to call it this way..
size_t sz;
params[0] = OSSL_PARAM_construct_size_t(OSSL_MAC_PARAM_SIZE, &sz);

Comment (slontis) [reply doc/man3/EVP_MAC.pod]: If that is not how we should call it then we need to get rid of all accessors that use size_t :)

Comment (paulidale) [reply doc/man3/EVP_MAC.pod]: We possibly need to discuss this disconnect -- I deliberated worded the man3 KDF pages in terms of the OSSL_PARAM types and the man7 pages in terms of the C types.  I don't think this is ideal.

Could we call it an _unsigned integer_ but add the expected maximum size in the description?
I.e. _The size of this parameter should not exceed that of B<size_t>._

Comment (levitte) [reply doc/man3/EVP_MAC.pod]: Please have a look at the discussion in #9875 and the resulting #9891

**Comment by slontis** [comment]
I have reverted some doc changes, since they have now been addressed in new PR's from @paulidale

**Comment by slontis** [comment]
ping: This is only a minor doc fix after rebasing..

**Comment by slontis** [comment]
ping

**Comment by slontis** [comment]
thanks.. merged to master.

**Pull request closed without merging**